tests/test_validation.py

```diff
@@ -576,34 +659,95 @@ def validate_hypotest(pdf, data, mu_test, expected_result, tolerance=1e-6):
         init_pars,
         par_bounds,
         return_expected_set=True,
-        test_stat="q",
+        test_stat=test_stat,
+        calctype=calctype,
     )
+
     assert abs(CLs_obs - expected_result['obs']) / expected_result['obs'] < tolerance
     for result, expected in zip(CLs_exp_set, expected_result['exp']):
         assert abs(result - expected) / expected < tolerance, result
 
 
 @pytest.fixture(
     params=[
-        ('1bin_shapesys', {'init_pars': 2, 'par_bounds': 2}, 1e-6),
-        ('1bin_lumi', {'init_pars': 2, 'par_bounds': 2}, 4e-6),
-        ('1bin_normsys', {'init_pars': 2, 'par_bounds': 2}, 2e-9),
-        ('2bin_histosys', {'init_pars': 2, 'par_bounds': 2}, 8e-5),
-        ('2bin_2channel', {'init_pars': 5, 'par_bounds': 5}, 1e-6),
-        ('2bin_2channel_couplednorm', {'init_pars': 2, 'par_bounds': 2}, 1e-6),
+        (
+            '1bin_shapesys',
+            {'init_pars': 2, 'par_bounds': 2},
+            1.0,
+            "q",
+            1e-6,
+            "asymptotics",
+        ),
+        (
+            '1bin_shapesys_q0',
+            {'init_pars': 2, 'par_bounds': 2},
+            0.0,
+            "q0",
+            3e-4,
+            "asymptotics",
+        ),
+        (
+            '1bin_shapesys_q0_toys',
+            {'init_pars': 2, 'par_bounds': 2},
+            0.0,
+            "q0",
+            1e-6,
+            "toybased",
+        ),
```

Comment:
This test is causing 

```python
assert abs(CLs_obs - expected_result['obs']) / expected_result['obs'] < tolerance
```

to fail in the `validate_hypotest` in `test_validation[1bin_shapesys_q0_mu1_toys]` given that

```
test_stat: q0                                                                                                                                                                                              
CLs_obs: 0.0035
obs expected results: 0.003
tolerance: 1e-06
abs difference: 0.0005
ratio: 0.16666666666666666
```

This is not related to the other JAX issues, as the results above comes from

```
$ pip list | grep jax
jax                           0.2.7
jaxlib                        0.1.57
```

@kratsg Can you set it up so that we can have the tests also accept a number of toys?